src/data/dfinsupp/lex.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2022 Damiano Testa and Junyan Xu. All rights reserved.
```

Comment:
```suggestion
Copyright (c) 2022 Junyan Xu. All rights reserved.
```
I *think* that the convention is that files have a unique copyright owner, but possibly several authors.

A similar comment applies to the `dne_locus` file.

Reply:
Thanks! I originally proved the well-foundedness result with finsupp, but wanted to make it as general as possible, but I noticed dfinsupp.lex doesn't exist, so I first did a minimal part of finsupp.lex to be able to state the result and port the well-founded proof. Then I realized I need ne_locus so I ported that file as well, and noticed it's actually quite smooth to port it completely since everything else is there, so I finished the dfinsupp.lex port after I finished the well-founded PR. Then I noticed I can golf finsupp using dfinsupp relatively easily, so that's a bonus :)

Reply:
Yes, it makes perfect sense to state stuff for `dfinsupp`.  However, I got here since I wanted results for `add_monoid_algebra`s so I felt already maxed out from the point of view of generalisations!  ☺️